invertedai/cosimulation.py

```diff
@@ -100,17 +115,32 @@ def ego_states(self):
         return [d for d, s in zip(self._agent_states, self._ego_agent_mask) if s]
 
     @property
-    def infractions(self) -> List[InfractionIndicators]:
+    def infractions(self) -> Optional[List[InfractionIndicators]]:
         """
         If `monitor_infractions` was set in the constructor,
         lists infractions currently committed by each agent, including ego agents.
         """
         return self._infractions
 
+    @property
+    def infraction_rates(self) -> Optional[Tuple[float, float, float]]:
```

Comment:
I'm not too keen on having this here. It doesn't seem particularly useful to me to average collisions across agents.